geoengine/layers.py

```diff
@@ -0,0 +1,660 @@
+LayerProviderId = NewType('LayerProviderId', UUID)
+
+LAYER_DB_PROVIDER_ID = LayerProviderId(UUID('ce5e84db-cbf9-48a2-9a32-d4b7cc56ea74'))
+LAYER_DB_ROOT_COLLECTION_ID = LayerCollectionId(UUID('05102bb3-a855-4a37-8a8a-30026a91fef1'))
+
+
+class LayerCollectionAndProviderIdResponse(TypedDict):
+    collectionId: str
+    providerId: str
+
+
+class LayerAndProviderIdResponse(TypedDict):
+    layerId: str
+    providerId: str
+
+
+class LayerCollectionResponse(TypedDict):
+    '''A layer collection response JSON from a HTTP request'''
+    id: LayerCollectionAndProviderIdResponse
+    name: str
+    description: str
+    items: List[LayerCollectionListingResponse]
+
+
+class LayerCollectionListingResponse(TypedDict):
+    '''A layer collection listing response JSON from a HTTP request'''
+    id: Union[LayerCollectionAndProviderIdResponse, LayerAndProviderIdResponse]
+    name: str
+    description: str
+    type: str
+
+
+class LayerCollectionListingType(LowercaseStrEnum):
+    LAYER = auto()
+    COLLECTION = auto()
+
+
+@dataclass(repr=False)
+class LayerCollectionListing:
+    '''A layer collection listing as item of a collection'''
+
+    listing_id: Union[LayerId, LayerCollectionId]
+    provider_id: LayerProviderId
+    name: str
+    description: str
+    type: LayerCollectionListingType
+
+    def __repr__(self) -> str:
+        '''String representation of a `LayerCollectionListing`'''
+
+        buf = StringIO()
+
+        buf.write(f"name: {self.name}{os.linesep}")
+        buf.write(f"description: {self.description}{os.linesep}")
+        buf.write(f"id: {self.listing_id}{os.linesep}")
+        buf.write(f"provider id: {self.provider_id}{os.linesep}")
+
+        return buf.getvalue()
+
+    def html_str(self) -> str:
+        '''HTML representation for Jupyter notebooks'''
+
+        buf = StringIO()
+
+        buf.write('<table>')
+        buf.write(f"<tr><th>name</th><td>{self.name}</td></tr>")
+        buf.write(f"<tr><th>description</th><td>{self.description}</td></tr>")
+        buf.write(f"<tr><th>id</th><td>{self.listing_id}</td></tr>")
+        buf.write(f"<tr><th>provider id</th><td>{self.provider_id}</td></tr>")
+        buf.write('</table>')
+
+        return buf.getvalue()
+
+    def _repr_html_(self) -> str:
+        '''HTML representation for Jupyter notebooks'''
+
+        return self.html_str()
+
+    def load(self,
+             offset: int = 0,
+             limit: int = 20,
+             timeout: int = 60) -> Union[LayerCollection, Layer]:
+        '''Load the listing item'''
+
+        if self.type == LayerCollectionListingType.COLLECTION:
+            return layer_collection(cast(LayerCollectionId, self.listing_id), self.provider_id, offset, limit, timeout)
+
+        if self.type == LayerCollectionListingType.LAYER:
+            return layer(cast(LayerId, self.listing_id), self.provider_id, timeout)
+
+        assert False, 'Invalid listing type'
+
+
+class LayerCollection:
```

Comment:
maybe store offset & limit?

Reply:
Now, we load everything:
[c6d8879](https://github.com/geo-engine/geoengine-python/pull/83/commits/c6d8879dbbde1ba5110530339e524e9f1e3a9ef2)